docker_images/ci-functional/Dockerfile

```diff
@@ -0,0 +1,84 @@
+FROM ubuntu:eoan
+
+LABEL maintainer="Conan.io <info@conan.io>"
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ENV PY27 /root/.pyenv/versions/2.7.17/bin/python
+ENV PY35 /root/.pyenv/versions/3.5.9/bin/python
+ENV PY37 /root/.pyenv/versions/3.7.6/bin/python
+ENV PY38 /root/.pyenv/versions/3.8.1/bin/python
+
+ENV CMAKE_3_16_4 /root/conan_packages/.conan/data/cmake/3.16.4/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/cmake
+ENV CMAKE_3_16_3 /root/conan_packages/.conan/data/cmake/3.16.3/_/_/package/44fcf6b9a7fb86b2586303e3db40189d3b511830/bin/cmake
+
+ENV GCC_7 /usr/bin/gcc-7
+ENV GXX_7 /usr/bin/g++-7
+ENV GCC_9 /usr/bin/gcc-9
+ENV GXX_9 /usr/bin/g++-9
+
+ENV CLANG_7 /usr/bin/clang-7
+ENV CLANGXX_7 /usr/bin/clang++-7
+ENV CLANG_9 /usr/bin/clang-9
+ENV CLANGXX_9 /usr/bin/clang++-9
+
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends \
+       sudo ca-certificates \
+       # pyenv requires
+       make build-essential libssl-dev zlib1g-dev libbz2-dev \
+       libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+       xz-utils tk-dev libffi-dev liblzma-dev python-openssl git \
+    \
+    # Install pyenv stuff
+    && curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash \
+    && export PATH="/root/.pyenv/bin:$PATH" \
+    && pyenv init - \
+    && pyenv virtualenv-init - \
+    && pyenv update \
+    && pyenv install 3.8.1 \
+    && pyenv install 3.7.6 \
+    && pyenv install 3.5.9 \
+    && pyenv install 2.7.17
+
+# Install CMake versions (using Conan) and register alternatives
+RUN $PY38 -m venv ./conan_tmp && cd ./conan_tmp \
+    && ./bin/pip install conan \
+    && export CONAN_USER_HOME=/root/conan_packages/ \
+    && ./bin/conan install cmake/3.16.4@ \
+    && ./bin/conan install cmake/3.16.3@ \
```

Comment:
2 versions?

Reply:
I'd like to add more than two versions, we may need to test Conan against different CMake versions, but right now we only have `3.16` is Conan Center 😞 

There are some meaningful versions we would like to test, Conan uses some features and sometimes we are not so good implementing that functionality with older versions.

Reply:
Of course, using here `3.16.3` and `3.16.4` is just an example

Reply:
But why, I mean, do you want to check behavior? just curious

Reply:
Eventually I would like to run the Conan test suite for different CMake versions, to check we do not rely on any new/old behavior. Have a look at these "docs" here: https://github.com/conan-io/conan_ci_jenkins/pull/10

Reply:
nice!

Reply:
Maybe now you understand my comment here: https://github.com/conan-io/conan-center-index/pull/633#issuecomment-573240765 😆